### DISCUSSION TITLE
Functional backend library

### DISCUSSION THREAD

**Discussion by rahul-neubla** [riscv-software-src/riscv-perf-model#68] [Q&A] [answered]
Hi,

I am looking to experiment with simple riscv cpu model using olympia and was wondering if there is any functional backend library that I can make use of (to run binary natively, instead of instr trace) ? Or if someone from riscv perf modelling community could suggest any open source library I can take a look at. 

Thanks,
Rahul

**Comment by klingaard** [comment, marked as answer]
There are plans to incorporate a Sparta version of a functional model, but the resources to make that happen are pretty tied up right now.  Of the open source libraries that I'm aware of for RISC-V co-simulation (or even verification) is [Esperanto's Dromajo functional simulator](https://github.com/chipsalliance/dromajo).  They have an API to use it as a library and integrate into co-simulation models.  That's something to get started with if you'd like to explore on your own...

**Reply by rahul-neubla** [reply]
Thanks, thats quite helpful. I will take a look at Dromajo.

**Comment by ghost** [comment]
Relates to #14
